pkg/gateway/model_build_targetgroup.go

```diff
@@ -188,27 +189,38 @@ func (t *latticeServiceModelBuildTask) buildTargets(ctx context.Context) error {
 	return nil
 }
 
-// TODO have a same BuildTargetGroup for both targetGroupModelBuildTask, latticeServiceModelBuildTask
-// Build target group for K8S serviceexport object
-func (t *targetGroupModelBuildTask) BuildTargetGroup(ctx context.Context) error {
+func (t *targetGroupModelBuildTask) BuildTargetGroupForServiceExport(ctx context.Context) error {
 	tgName := latticestore.TargetGroupName(t.serviceExport.Name, t.serviceExport.Namespace)
+	var tg *latticemodel.TargetGroup
+	var err error
+	if t.serviceExport.DeletionTimestamp.IsZero() {
+		tg, err = t.buildTargetGroupForServiceExportCreation(ctx, tgName)
+	} else {
+		tg, err = t.buildTargetGroupForServiceExportDeletion(ctx, tgName)
+	}
+	if err != nil {
+		return err
+	}
+	t.tgByResID[tgName] = tg
+	t.targetGroup = tg
+	return nil
+}
 
+func (t *targetGroupModelBuildTask) buildTargetGroupForServiceExportCreation(ctx context.Context, targetGroupName string) (*latticemodel.TargetGroup, error) {
 	svc := &corev1.Service{}
 	if err := t.client.Get(ctx, k8s.NamespacedName(t.serviceExport), svc); err != nil {
-		// mark there is no serviceexport dependence on the TG
-		t.datastore.SetTargetGroupByServiceExport(tgName, false, false)
-		return fmt.Errorf("error finding corresponding service %v error :%w", k8s.NamespacedName(t.serviceExport), err)
+		t.datastore.SetTargetGroupByServiceExport(targetGroupName, false, false)
+		return nil, fmt.Errorf("Failed to find corresponding k8sService %s error :%w \n", k8s.NamespacedName(t.serviceExport).String(), err)
```

Comment:
Why are trailing `\n`s added after the change? (on many lines including this) Can we remove them?